sechub-pds-tools/src/main/java/com/mercedesbenz/sechub/pds/tools/PDSToolsCLI.java

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.pds.tools;
+
+import java.io.File;
+
+import com.mercedesbenz.sechub.pds.tools.generator.PDSSolutionTestFilesGenerator;
+import com.mercedesbenz.sechub.pds.tools.handler.ConsoleHandler;
+import com.mercedesbenz.sechub.pds.tools.handler.ExitHandler;
+import com.mercedesbenz.sechub.pds.tools.handler.PrintStreamConsoleHandler;
+import com.mercedesbenz.sechub.pds.tools.handler.SystemExitHandler;
+
+public class PDSToolsCLI {
+
+    ConsoleHandler consoleHandler;
+    ExitHandler exitHandler;
+
+    public PDSToolsCLI() {
+        consoleHandler = new PrintStreamConsoleHandler();
+        exitHandler = new SystemExitHandler();
+    }
+
+    public static void main(String[] args) throws Exception {
+        PDSToolsCLI cli = new PDSToolsCLI();
+        cli.start(args);
+    }
+
+    void start(String... args) throws Exception {
+        if (args.length == 0) {
+            showHelpAndExit("Arguments missing", 1);
+        }
+        String command = args[0];
+        switch (command) {
+
+        case "--help":
+            showHelp();
+            exitHandler.exit(0);
+        case "--generate":
+            PDSSolutionTestFilesGenerator generator = new PDSSolutionTestFilesGenerator();
+            generator.setOutputHandler(consoleHandler);
+
+            if (args.length < 3 || args.length > 4) {
+                showHelpAndExit("Generate command needs 2 additional parameters: 1. config file path, 2.scan type", 3);
+            }
+            String pathToConfigFile = args[1];
+            String scanType = args[2];
+            File targetFolder = null;
+            if (args.length == 4) {
+                String targetFolderPath = args[3];
+                targetFolder = new File(targetFolderPath);
+                if (!targetFolder.exists()) {
+                    targetFolder.mkdirs();
+                }
+            }
+            generator.generate(pathToConfigFile, scanType, targetFolder);
+
+            break;
+
+        default:
+            showHelpAndExit("Unrecognized command:" + command, 2);
+        }
+    }
+
+    private void showHelpAndExit(String message, int exitCode) {
+        consoleHandler.output("Wrong usage:" + message);
+        showHelp();
+        exitHandler.exit(exitCode);
+    }
+
+    private void showHelp() {
+        consoleHandler.output("Usage:");
+        consoleHandler.output("--help                                          ");
+        consoleHandler.output("   show this output");
+        consoleHandler.output("");
+        consoleHandler.output("--generate ${secHubConfigFilePath} ${scanType} [$targetFolderPath] ");
+        consoleHandler.output("   generate PDS test files for given config. Given scan type");
+        consoleHandler.output("   can be: codeScan, licenseScan, webScan etc. When no target folder");
```

Comment:
Is there no Enum for the types? An enum which makes it possible to simply list the possible values of the enum. In addition, one could check whether the user used an existing value.

Reply:
I changed the output to following style (and it does now use enum etc.)

```
SecHub PDS tools CLI
--------------------
Usage:
--help
   Show this help output.
--generate ${secHubConfigFilePath} ${scanType} [${targetFolderPath}]
   Generate PDS test files.
     - secHubConfigFilePath
        path to the sechub config file used to generate.
     - scanType
        scan type - must be one of: codeScan, webScan, infraScan, licenseScan.
     - targetFolderPath (optional)
        When not defined, a temp folder will be created and used.

```